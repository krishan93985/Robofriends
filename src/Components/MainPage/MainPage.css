@font-face {
    font-family: 'SEGA LOGO FONT';
    font-style: normal;
    font-weight: normal;
    font-display:swap;
    src: local('SEGA LOGO FONT'), url('../../SEGA.woff') format('woff');
    }

h1{
    font-family:'SEGA LOGO FONT','monospace';
    color:#0ccac4;
    font-weight: 200;
}
