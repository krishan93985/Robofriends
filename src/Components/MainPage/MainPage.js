import React from "react";
import Header from "../Header/Header";
import Cardarray from "../Cardarray/Cardarray";
import Searchbox from "../Searchbox/Searchbox";
import Scroll from "../Scroll/Scroll";
import "./MainPage.css";

class MainPage extends React.Component {
  componentDidMount() {
    this.props.fetchRobots();
  }

  filterRobots = () => {
    const { robots, searchField } = this.props;
    return robots.filter((robot) => {
              return robot.username.toLowerCase().includes(searchField.toLowerCase());
           });
  }

  render() {
    const { onSearchChange, isPending } = this.props;

    return (
      <div className="tc">
        <Header />
        <Searchbox onsearch={onSearchChange} />
        <Scroll>
          {isPending ? (
            <h1 id="loader" className="tc">Loading...</h1>
          ) : (
            <Cardarray robots={this.filterRobots()} />
          )}
        </Scroll>
      </div>
    );
  }
}

export default MainPage;
