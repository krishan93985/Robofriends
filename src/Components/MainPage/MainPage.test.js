import React from 'react';
import MainPage from './MainPage';
import { shallow, mount, render } from 'enzyme';
import toJson from 'enzyme-to-json';

let renderWrapper,filterWrapper,pendingWrapper;

beforeEach(() => {
    const mockRenderProps = {
        onSearchChange:jest.fn(),
        fetchRobots:jest.fn(),
        searchField:'',
        robots:[],
        isPending:false,
        error:''
    }
    const mockFilterProps = {
        fetchRobots:jest.fn(),
        searchField:'jo',
        robots:[{
            id:2,
            name:'John',
            username:'John123',
            email:'John123@gmail.com'
        }],
        isPending:false,
    }
    const mockPendingProps = {
        fetchRobots:jest.fn(),
        searchField:'',
        robots:[],
        isPending:true,
    }
    renderWrapper = shallow(<MainPage {...mockRenderProps}/>)
    filterWrapper = shallow(<MainPage {...mockFilterProps}/>)
    pendingWrapper = shallow(<MainPage {...mockPendingProps}/>)
})

it('MainPage component renders correctly', () => {
    expect(toJson(renderWrapper)).toMatchSnapshot();
})

it('filters robots correctly', () => {
    expect(filterWrapper.instance().filterRobots()).toEqual([{
        id:2,
        name:'John',
        username:'John123',
        email:'John123@gmail.com'
    }]);
})

it('renders pending state correctly', () => {
    expect(pendingWrapper.find("[id='loader']").length).toEqual(1);
})