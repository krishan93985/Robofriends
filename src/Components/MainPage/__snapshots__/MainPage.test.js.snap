// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainPage component renders correctly 1`] = `
<div
  className="tc"
>
  <Memo() />
  <SearchBox
    onsearch={[MockFunction]}
  />
  <Scroll>
    <Cardarray
      robots={Array []}
    />
  </Scroll>
</div>
`;
